### ISSUE TITLE
Select Top 1000 Records Stops working

### ISSUE THREAD

**Issue by jtsom** [microsoft/azuredatastudio#23414]
Type: <b>Bug</b>

Right clicking on a table and selecting "Select Top 100 Records" only works the first time. Any subsequent attempts to use this command, on any table, does nothing. No response.

This is the developer tools logs after running the command the first, successful, time:

```
[Extension Host] (node:19636) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `azuredatastudio-insiders --trace-deprecation ...` to show where the warning was created) (at c:\Users\jtsombakos\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:83:196474)
C @ console.ts:137
$logExtensionHostMessage @ mainThreadConsole.ts:39
_doInvokeHandler @ rpcProtocol.ts:457
_invokeHandler @ rpcProtocol.ts:442
_receiveRequest @ rpcProtocol.ts:370
_receiveOneMessage @ rpcProtocol.ts:296
(anonymous) @ rpcProtocol.ts:161
invoke @ event.ts:622
deliver @ event.ts:824
fire @ event.ts:785
fire @ ipc.net.ts:638
_receiveMessage @ ipc.net.ts:959
(anonymous) @ ipc.net.ts:832
invoke @ event.ts:622
deliver @ event.ts:824
fire @ event.ts:785
acceptChunk @ ipc.net.ts:382
(anonymous) @ ipc.net.ts:338
_ @ ipc.net.ts:70
emit @ node:events:526
addChunk @ node:internal/streams/readable:315
readableAddChunk @ node:internal/streams/readable:289
Readable.push @ node:internal/streams/readable:228
onStreamRead @ node:internal/stream_base_commons:190
log.ts:301  INFO Disconnect request sent for connection:providerName:MSSQL|authenticationType:Integrated|database:DigitalOptometrics_Auditing|server:usma8593\sqlexpress|user:|databaseDisplayName:DigitalOptometrics_Auditing|group:ffc8098d-5ec3-4f0b-9d4f-f64e8db56fd2 - deleting connection
log.ts:301  INFO Deleting connection connection:providerName:MSSQL|authenticationType:Integrated|database:DigitalOptometrics_Auditing|server:usma8593\sqlexpress|user:|databaseDisplayName:DigitalOptometrics_Auditing|group:ffc8098d-5ec3-4f0b-9d4f-f64e8db56fd2
log.ts:301  INFO Disconnect request sent for untitled:SQLQuery_2 - deleting connection
log.ts:301  INFO Deleting connection untitled:SQLQuery_2
log.ts:301  INFO Removing connection connection:providerName:MSSQL|authenticationType:Integrated|database:DigitalOptometrics_Auditing|server:usma8593\sqlexpress|user:|databaseDisplayName:DigitalOptometrics_Auditing|group:ffc8098d-5ec3-4f0b-9d4f-f64e8db56fd2
log.ts:313   ERR Cannot read properties of undefined (reading 'savePassword'): TypeError: Cannot read properties of undefined (reading 'savePassword')
    at q.disconnectConnection (vscode-file://vscode-app/c:/Users/jtsombakos/AppData/Local/Programs/Azure%20Data%20Studio%20-%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:2890:41423)
    at vscode-file://vscode-app/c:/Users/jtsombakos/AppData/Local/Programs/Azure%20Data%20Studio%20-%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:2890:35010
    at h.invoke (vscode-file://vscode-app/c:/Users/jtsombakos/AppData/Local/Programs/Azure%20Data%20Studio%20-%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:169:145)
    at u.deliver (vscode-file://vscode-app/c:/Users/jtsombakos/AppData/Local/Programs/Azure%20Data%20Studio%20-%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:169:2266)
    at r.fire (vscode-file://vscode-app/c:/Users/jtsombakos/AppData/Local/Programs/Azure%20Data%20Studio%20-%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:169:1844)
    at ba._notifyDisconnected (vscode-file://vscode-app/c:/Users/jtsombakos/AppData/Local/Programs/Azure%20Data%20Studio%20-%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:2712:44065)
    at vscode-file://vscode-app/c:/Users/jtsombakos/AppData/Local/Programs/Azure%20Data%20Studio%20-%20Insiders/resources/app/out/vs/workbench/workbench.desktop.main.js:2712:40643
    at _ZoneDelegate.invoke (C:\Users\jtsombakos\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\node_modules.asar\zone.js\dist\zone.js:409:30)
    at Zone.run (C:\Users\jtsombakos\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\node_modules.asar\zone.js\dist\zone.js:169:47)
    at C:\Users\jtsombakos\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\node_modules.asar\zone.js\dist\zone.js:1326:38
    at _ZoneDelegate.invokeTask (C:\Users\jtsombakos\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\node_modules.asar\zone.js\dist\zone.js:443:35)
    at Zone.runTask (C:\Users\jtsombakos\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\node_modules.asar\zone.js\dist\zone.js:214:51)
    at drainMicroTaskQueue (C:\Users\jtsombakos\AppData\Local\Programs\Azure Data Studio - Insiders\resources\app\node_modules.asar\zone.js\dist\zone.js:632:39)
DevTools failed to load source map: Could not load content for https://sqlopsbuilds.blob.core.windows.net/sourcemaps/cdece24123139285de21ab25e9281786fca7d93b/node_modules/@vscode/vscode-languagedetection/dist/lib/index.js.map: Fetch through target failed: Target not supported; Fallback: Unknown error
```

Azure Data Studio version: azuredatastudio-insiders 1.45.0-insider (cdece24123139285de21ab25e9281786fca7d93b, 2023-06-19T05:36:45.774Z)
OS version: Windows_NT x64 10.0.22621
Restricted Mode: No
Preview Features: Enabled
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8500 CPU @ 3.00GHz (6 x 3000)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|31.78GB (16.21GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (7)</summary>

Extension|Author (truncated)|Version
---|---|---
dacpac|Mic|1.13.0
import|Mic|1.5.5
schema-compare|Mic|1.19.0
sql-search|Red|0.4.0
ADS-add-columns|sof|0.0.12
vscodeintellicode|Vis|1.2.1
poor-sql-formatter|WSR|0.1.0

</details>

**Comment by alanrenmsft** [comment]
@jtsom I am not able to reproduce the issue, what type of server were you using? Was there a network outage at that time?

**Comment by jtsom** [comment]
I am still able to reproduce this error. In this case, I am using my local SQLEXPRESS dataabase, so network is not an issue.

What I just did:
1.  launched ADS (version info below)
2.  clicked on my database
3.  clicked to open the tables
4.  right clicked a table. chose Show top 1000 records
5. Records are shown in the result area
6. Picked another table, right clicked Show top 1000 records.
7. Nothing happens.

I now need to quit and relaunch ADS to be able to do that.

Note; I can still enter direct SQL queries and they run - I can also click the Run button in the "show top 1000 records" window and the query will re-run.

Version info:
```
Version: 1.45.0-insider (user setup)
Commit: cdece24123139285de21ab25e9281786fca7d93b
Date: 2023-06-19T05:36:45.774Z
VS Code: 1.70.0
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.22621
```

**Comment by alanrenmsft** [comment]
@jtsom thanks, let me try again

**Comment by jtsom** [comment]
I just noticed, after doing the initial Show top 1000 - none of the right-click options in the table / database list work (Script as create, etc.)

**Comment by alanrenmsft** [comment]
@jtsom do you happen to have other servers you can use? just want to make sure it is not sql express specific.

**Comment by jtsom** [comment]
Not sure if it's related, but right-clicking on a table (view, etc) and choosing "Refresh" generates an error notification, with this in the developer console.

![image](https://github.com/microsoft/azuredatastudio/assets/13488/abb20492-35fd-454e-975b-0865f99666f9)

**Comment by jtsom** [comment]
I am checking one of our Azure SQL databases (but it may take a while, as using the Azure VPN + ADS is HORRIBLY show to not useable - there is an open, active, item for this issue https://github.com/microsoft/azuredatastudio/issues/7440).

**Comment by jtsom** [comment]
(that was painful - Azure VPN + FQDN to server + ADS is all but unusable)

Either way, after several connect-disconnect-connect cycles, I was able to test, and yes it is doing the same thing. First execution works, returns a response. Thereafter, nothing.

**Comment by jtsom** [comment]
[20230620T150705.zip](https://github.com/microsoft/azuredatastudio/files/11805720/20230620T150705.zip)
[Trace.txt](https://github.com/microsoft/azuredatastudio/files/11805722/Trace.txt)

**Comment by alanrenmsft** [comment]
from the log, it seems the second script never reached the backend service side. will continue investigating the issue.

btw, did you do something else after clicking the `Select top 1000` menu item?

**Comment by jtsom** [comment]
No, I just tried to execute the command again. It looks like once a command is chosen and executed from the context menu, any other attempts are ignored.

**Comment by alanrenmsft** [comment]
do you have this setting on? https://github.com/microsoft/azuredatastudio/issues/7440#issuecomment-1148069779, if not, could you please give it a try?

**Comment by jtsom** [comment]
I have that setting on. I tried turning off (and restarting), but am getting the same incorrect behavior.

**Comment by jtsom** [comment]
FWIW, latest Insider
```
Version: 1.45.0-insider (user setup)
Commit: b028d1257772aa8c1d73b316a9bddfd7e9c6c00b
Date: 2023-06-21T06:00:57.682Z
VS Code: 1.70.0
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.22621
```
still has the problem.

**Comment by alanrenmsft** [comment]
@jtsom My change was not included in last night's build, I just published another insiders build, please upgrade and try the scenario again, Now you should be able to tell whether it is in progress and error will be displayed if something went wrong.

**Comment by jtsom** [comment]
Updated.
Running the command the second time, and there's an error notification:

![image](https://github.com/microsoft/azuredatastudio/assets/13488/c6bf87fc-4425-4f47-bcd5-92e7f0e10b27)

Adding some breakpoints in the developer console, the error is being thrown at line 223:

![image](https://github.com/microsoft/azuredatastudio/assets/13488/9b0d9cc0-05c5-4ab5-9e6e-23b59f9e2817)

**Comment by alanrenmsft** [comment]
@jtsom thanks.

@aasimkhan30 , could you please take a look at the issue?

**Comment by jtsom** [comment]
FYI, latest insiders, no change
```
Version: 1.45.0-insider (user setup)
Commit: 9781a6c72af47472e409a161b9f36b4a9e8f0f19
Date: 2023-06-22T05:58:57.303Z
VS Code: 1.70.0
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.22621
```

Also, just tested. It's not limited to executing the Select Top 1000" command. Using ANY command in the context menu (Script as Create, etc.) will cause subsequent commands to fail.

**Comment by jtsom** [comment]
Also, the notification that shows the error (screenshot above), does not go away. It has the "status line" constantly moving, with no way to close it. Clicking the "Bell" icon will hide it, but it still shows as "not acknowledged."

**Comment by alanrenmsft** [comment]
> Also, the notification that shows the error (screenshot above), does not go away. It has the "status line" constantly moving, with no way to close it. Clicking the "Bell" icon will hide it, but it still shows as "not acknowledged."

thanks for the feedback, I've fixed this just now, in case of error, the in progress indicator will be removed and a close button is added so you can close it.

![image](https://github.com/microsoft/azuredatastudio/assets/13777222/7b27745a-9afc-4bec-a6f7-fea4bc8aa6ba)


@aasimkhan30 will look into the actual error here, he will let you know when the issue is fixed.

**Comment by jtsom** [comment]
@aasimkhan30 any progress on this? It's making using ADS difficult at best.

Still an issue in latest version:
Version: 1.45.0-insider (user setup)
Commit: 030a05b7ffd448ffc26ff5cd55480cf1f2ef5d7f
Date: 2023-06-28T14:14:54.654Z
VS Code: 1.79.0
Electron: 22.3.10
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.22621

**Comment by cheenamalhotra** [comment]
Related with https://github.com/microsoft/azuredatastudio/issues/23501, this is being investigated.
